src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs

```diff
@@ -41,5 +41,25 @@ public static IWebHostBuilder UseSerilog(this IWebHostBuilder builder, Serilog.I
                 collection.AddSingleton<ILoggerFactory>(new SerilogLoggerFactory(logger, dispose)));
             return builder;
         }
+
+        /// <summary>Sets Serilog as the logging provider.</summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="configureSerilog">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Logger" />.</param>
+        /// <returns>The web host builder.</returns>
+        public static IWebHostBuilder UseSerilog(this IWebHostBuilder builder, Action<WebHostBuilderContext, LoggerConfiguration> configureSerilog)
+        {
+            if (builder == null) throw new ArgumentNullException(nameof(builder));
+            if (configureSerilog == null) throw new ArgumentNullException(nameof(configureSerilog));
+            builder.ConfigureServices
+            (
+                (context, collection) =>
+                {
+                    var loggerConfiguration = new LoggerConfiguration();
+                    configureSerilog(context, loggerConfiguration);
+                    collection.AddSingleton<ILoggerFactory>(new SerilogLoggerFactory(loggerConfiguration.CreateLogger(), true));
```

Comment:
Does `AddSingleton` guarantee disposal of the logger factory?

Reply:
I think the created `ILogger` should to be saved to the static `Log.Logger` here, because at some point in the `Startup.Configure()` function, the `CloseAndFlush()` function needs to be called:

```cs
var applicationLifetime = app.ApplicationServices.GetRequiredService<IApplicationLifetime>();
applicationLifetime.ApplicationStopping.Register(() => Log.CloseAndFlush());
```

Reply:
Can we call those "sensible defaults" and just start out by wiring all of that up automatically here? The overall perspective would then be: _if you supply the logger, you configure everything to your liking; if we supply it (via this new overload), "just trust us", everything should be set up reasonably._ Does that make sense?

Reply:
According to [the documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection#disposing-of-services), it seems AddSingleton won't dispose the logger factory if we create the object ourselves. 😟
I think that using the overload with the factory method should do the trick, however.

Regarding the call to `CloseAndFlush()` method, should I try to wire it in the call to `ConfigureServices()`, then ?